refactor(WIP): gnoland home realm

alexiscolin opened the pull request:
Related to #3249,  This PR init the refactor of the gno.land homepage, focusing on content sections and design.

Related: [Figma design file](https://www.figma.com/design/bWAWFaIFtvFISqZCzvJ2ID/gno.land---Realm?node-id=1-27&t=66DaWFccaPMgTskM-0).

TODO:
- [x] Columns are implemented in code but require #3763 to be merged in order to appear in the UI.
- [ ] Images need to be added to the assets folder (WIP).
- [ ] Finalized content must be added.
- [ ] Once `Columns` component is merged, we should add a new `Input` component to manage the airdrop tracker.
- [ ] Add meta title and description


⚠️  Work in progress (WIP) with placeholder text, and the correct content still needs to be provided by the relevant counterparts in this PR.

alexiscolin committed:
refactor: gnoland home realm

alexiscolin committed:
Merge branch 'master' into refactor/gnoland-home

Gno2D2 commented:
#### 🛠 PR Checks Summary
All **Automated Checks** passed. ✅

##### Manual Checks (for Reviewers):
- [ ] **IGNORE** the bot requirements for this PR (force green CI check)


<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.

##### ✅ Automated Checks (for Contributors):
 🟢 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))


##### ☑️ Contributor Actions:
1. Fix any issues flagged by automated checks.
2. Follow the Contributor Checklist to ensure your PR is ready for review.
	- Add new tests, or document why they are unnecessary.
	- Provide clear examples/screenshots, if necessary.
	- Update documentation, if required.
	- Ensure no breaking changes, or include `BREAKING CHANGE` notes.
	- Link related issues/PRs, where applicable.

##### ☑️ Reviewer Actions:
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.

##### 📚 Resources:
- [Report a bug with the bot](https://www.github.com/gnolang/gno/issues/3238).
- [View the bot’s configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 And
    ├── 🟢 The base branch matches this pattern: ^master$
    └── 🟢 The pull request was created from a fork (head branch repo: alexiscolin/gno)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

</blockquote></details>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>**IGNORE** the bot requirements for this PR (force green CI check)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Can be checked by

- Any user with comment edit permission

</blockquote></details>

</blockquote></details>

</blockquote></details>

</details>

alexiscolin commented:
@leon, should we have another way to update the content than by using `AdminSetOverride` (I ask for the future)?

alexiscolin commented:
> @leohhhn, should we consider an alternative method to AdminSetOverride for updating content, with future scalability in mind?

I guess it should be ok, since Manfred updated the realm one week ago with this feature.

michelleellen commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,220 @@ func Render(_ string) string {
-Intuitive and easy to use, gno.land lowers the barrier to web3 and makes
-censorship-resistant platforms accessible to everyone. If you want to help lay
-the foundations of a fairer and freer world, join us today. 
+Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus ultricies ultricies risus. Cras libero est, congue vitae ex nec, sodales rutrum leo. Quisque a blandit ex, at viverra ex.Donec nulla risus, ullamcorper rutrum sapien ut, varius interdum quam. Donec non sapien varius, iaculis nibh at, placerat risus. 
```
Suggestion:
We’re building gno.land, set to be the leading open-source smart contract platform for the Internet of the future. At its core, gno.land is powered by Gno, an interpreted and fully deterministic variation of the Go programming language, designed for succinct and composable smart contracts.

The gno.land blockchain is the first experiment showcasing what’s possible with Gno, providing a secure, auditable, and reusable architecture for building and iterating on decentralized applications. 
With transparent and timeless code, gno.land serves as the “GitHub” of the Gno ecosystem, enabling anyone to inspect, reuse, and contribute to the development of Gno-based dApps.

Intuitive and easy to use, gno.land makes anti-censorship-resistant web3 applications the standard for the future. Join us in building an open-source-first, fairer, and freer Internet for everyone

Note: we might want to bring in the DAO component to the intro blurb?

alexiscolin replied:
Maybe in your review, you could commit the texts at the exact place in order to replace the placeholders (lorem ipsum) @michelleellen ? I mean line by line. That would help tracking the content and ensuring everything is setup and you don't miss anything.

michelleellen committed:
Update home.gno

Proposed suggestions for content for the homepage to be reviewed and iterated upon.

moul commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,233 @@ func Render(_ string) string {
 		out, _ := events.RenderEventWidget(events.MaxWidgetSize)
 		return out
 	})
-	r.RegisterCallback(":latest-hof:", func() string {
-		return hof.RenderExhibWidget(5)
-	})
-	r.RegisterCallback(":qotb:", quoteOfTheBlock)
-	r.RegisterCallback(":chain-height:", func() string {
-		return strconv.Itoa(int(std.ChainHeight()))
-	})
```
you should not remove these things, they are about allowing to have dynamic content on the homepage which is a target.

ps: you should also use some of them in the template code.

alexiscolin replied:
@moul We plan to use them (dynamic parts) for other purpose (roadmap for ex).

moul replied:
I think you didn't get it. we NEED to keep this.

leohhhn replied:
So the idea would be to have these sections be dynamic, ie using the hall of fame & blog realms to get dynamic data. @michelleellen, let's add the HoF, latest events, and latest blogs to the new home page. Then we can make them actually dynamic within code.

michelleellen replied:
For blog and events we have them in the community of gnomes section, these are dynamic right? https://github.com/gnolang/gno/pull/3888/files#r2014740015

So then it would be just about adding HoF to the builders section as dynamic data? 

Ideally you navigate from the builders section to a builders realm/page, that has more dynamic data such as new packages, examples, awesome gno, tutorials, r/docs..etc. Or do we want that on the home page in the builders section which right now highlights docs, github, tutorials, grants and bounties...

If possible a dynamic data section with the project state would be great. I think at one point Manfred mentioned that if people arrived at the homepage they should be able to see dynamically the current 'state' of the project, is this correct?

alexiscolin replied:
@leohhhn As @michelleellen said, the blog and event sections are still in the realm and of course are still dynamic 👍

alexiscolin replied:
I let you decide what to do with the other dynamic sections we previously removed from the current page: `hof`, `block of the day` etc.

moul commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,233 @@ func Render(_ string) string {
-We’re building gno.land, set to become the leading open-source smart contract
-platform, using Gno, an interpreted and fully deterministic variation of the
-Go programming language for succinct and composable smart contracts.
+## We're building a user-owned internet where governance is fair, development is open, and innovation is rewarded.
```
we should not use H2 just to make text bigger.

alexiscolin replied:
I think it's a good idea to use it as an H2 for SEO purposes (not for text size).

moul commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,233 @@ func Render(_ string) string {
-:latest-hof:
+</gno-columns>
+
+TODO: Add 'Beta Mainnet Coming Soon!' section
```
@leohhhn can you make this dynamic, i.e., by having like r/gnoland/blog to have a "news" tag?

moul commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,233 @@ func Render(_ string) string {
+
+TODO: Include the gno.land roadmap dynamic infographic
+
+**A moment in time (May 20th, 2020)*: Cosmos snapshot 
```
```suggestion
**A moment in time (May 20th, 2022)*: Cosmos snapshot 
```

iirc

michelleellen committed:
Update examples/gno.land/r/gnoland/home/home.gno

Co-authored-by: Manfred Touron <94029+moul@users.noreply.github.com>

michelleellen committed:
Update home.gno

michelleellen commented:
Removed the TODO content and this content from the PR, we can add it in later 

**A moment in time (May 20th, 2022)*: Cosmos snapshot 
**Milestone complete (July 2024)**: A multi-node, permanent and stable testnet with 1st implementation of the GovDAO
**Milestone complete (November 2024)**: Extended validator set, testnet improvements and GovDAO v2
**Milestone 'Beta Mainnet Release' (March 2025)**: GovDAO v3, r/Boards, Token distribution, Security audit complete
**Milestone 'Build the Future' (2025)**: Bridging Cosmos and gno.land, the foundations of a network of everyone
**Milestone 'Mainnet' (2025)**: A complete mainnet with token transfers and Cosmos bridge with ICS and IBC

moul reviewed (changes requested):
(just blocking because I need to approve this one)

michelleellen committed:
Update home.gno

content update March 20th

alexiscolin commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,238 @@ func Render(_ string) string {
+A decentralized space for uncensored discourse, fair moderation, and community-driven governance with forkability to allow for conflict resolution.
+
+
+####
```
```suggestion
```

alexiscolin committed:
Update examples/gno.land/r/gnoland/home/home.gno

Kouteki commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,237 @@ func Render(_ string) string {
-the foundations of a fairer and freer world, join us today. 
 
-## Learn about gno.land
+**/about /github /try gno**
```
Resolves to:

`https://home.gnoteam.com/github`
`https://home.gnoteam.com/try-gno`

Kouteki commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,237 @@ func Render(_ string) string {
+
+### Our Blog 
+
+:latest-blogposts:
```
Returns `No posts.`

leohhhn replied:
This data is dynamic from the gnoland/blog realm

Kouteki commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,237 @@ func Render(_ string) string {
+
+![Events](#)
+
+### Past Events
```
Returns `No events.`

leohhhn replied:
This data is dynamic from the gnoland/events realm

Kouteki commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,237 @@ func Render(_ string) string {
+
+gno.land builders are at the center of creating the applications and infrastructure for the decentralized operating system needed to power an Internet that is 100% decentralized.
+
+[/read the documentation](docs.gno.land) [Read about the Gno language](https://gno.land/gnolang)
```
```suggestion

- [read the documentation](https://docs.gno.land)
- [Read about the Gno language](https://gno.land/gnolang)

```

Kouteki replied:
The 2nd link could be resolved as a gno.land page

Kouteki commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,237 @@ func Render(_ string) string {
-> :qotb:
+The Gno bounty program is a good way to find interesting challenges in Gno, and get rewarded for helping us advance the project. 
+
+[/read more](https://github.com/gnolang/bounties)
```
```suggestion
[Read more](https://github.com/gnolang/bounties)
```

Kouteki commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,237 @@ func Render(_ string) string {
+
+Are you a builder, tinkerer, or researcher? If you’re looking to create awesome dApps, tooling, infrastructure, or smart contract libraries on gno.land, you can apply for a grant. gno.land grants program provides financial support to developers interested in exploring new blockchain technology, and/or are already actively contributing to gno.land.
+
+[/read more](https://github.com/gnolang/grants)
```
```suggestion
[Read more](https://github.com/gnolang/grants)
```

michelleellen commented:
We need to add the email subscription to the page as well @alexiscolin

moul commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,237 @@ func Render(_ string) string {
-We’re building gno.land, set to become the leading open-source smart contract
-platform, using Gno, an interpreted and fully deterministic variation of the
-Go programming language for succinct and composable smart contracts.
+## We’re building a decentralized operating system where governance is fair, development is open, and innovation is rewarded.
```
We don't reward innovation; we reward "making good things," even if they are boring, as long as they are useful. 

This phrase is excellent—one of the best we've had—but we need to refine it. I suggest we hold a mini brainstorming session during a meeting.

moul commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,237 @@ func Render(_ string) string {
-With transparent and timeless code, gno.land is the next generation of smart
-contract platforms, serving as the “GitHub” of the ecosystem, with realms built
-using fully transparent, auditable code that anyone can inspect and reuse.
+[/about](/about) &nbsp;&nbsp;&nbsp; [/github](/github) &nbsp;&nbsp;&nbsp; [/try gno](/try-gno) 
```
Can we create an ecosystem where `&nbsp;` doesn't exist?

moul commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,237 @@ func Render(_ string) string {
-:upcoming-events:
+gno.land’s smart contract platform is a first step in an experiment to build a complete dOS to bypass the centralized and controlled restrictions of the current Web, and power new social platforms, governance frameworks, and economic models that favor contribution-based incentives.
+
+<gno-columns>
```
I think we should avoid those columns until we have at least one link for each of them. Let's focus on using efficient paragraphs. Columns will be used not just to display information differently, but also to invite people to choose what they want to explore in depth.

moul commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,237 @@ func Render(_ string) string {
+</gno-columns>
 
 ---
+Beta Mainnet
```
example of news section that should be dynamic.

moul commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,237 @@ func Render(_ string) string {
+
+Explore gno.land's expanding ecosystem of applications, wallets, and explorers.
+
+[/the ecosystem ](https://gno.land/ecosystem)
```
```suggestion
[the ecosystem ](/ecosystem)
```

only use relative links.

moul commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,237 @@ func Render(_ string) string {
+## Beta Mainnet Coming Soon!
+
+Stay tuned for the first phase of gno.land's launch, the beta mainnet release.  
+- [Road to Mainnet and Beyond](https://gno.land/r/gnoland/blog:p/road-to-mainnet)
```
```suggestion
- [Road to Mainnet and Beyond](/r/gnoland/blog:p/road-to-mainnet)
```

only relative links

moul commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,237 @@ func Render(_ string) string {
+
+![Wallet](#)
+
+### Wallets and Faucet
+
+Execute transactions on gno.land using wallets, or a CLI keychain and client tool.
```
Leon, I think we have better content on docs.gno.land. Let's keep the homepage light while inviting visitors to go to docs.gno.land for more information about the tools.

moul commented on examples/gno.land/r/gnoland/home/home.gno:
```diff
@@ -30,144 +25,237 @@ func Render(_ string) string {
+## Why Gno and gno.land?
 
-:upcoming-events:
+gno.land’s smart contract platform is a first step in an experiment to build a complete dOS to bypass the centralized and controlled restrictions of the current Web, and power new social platforms, governance frameworks, and economic models that favor contribution-based incentives.
```
2cts: We have an "About" link, followed by an introductory paragraph and here the "why gno" paragraph. 

Could we try create a single more efficient introductory paragraph or set of paragraphs with a link to learn more (/about)?

jefft0 commented:
@alexiscolin, this PR has been idle for a year. Do you expect do revive it so that it is part of the user onboarding experience? Is there another PR for the gnoland home realm which you feel is an update to this PR?
